config/routes.rb

```diff
@@ -2,6 +2,17 @@
   crud = %i(index show create update destroy)
 
   namespace :v1 do
-    resources :projects, only: crud
+    resources :projects, only: crud do
+      post :create_task
+
+      member do
+        get :list_tasks
+      end
+
+    resources :tasks, only: [:update] do
+      put :update_task_state
+    end
```

Comment:
I approached it from a pretty straightforward perspective, “I want to GET all of my tasks from a specific project. I want to POST a new task to a specific project. I want to UPDATE a specific task.” Originally, I wanted to see if I could handle the endpoints only in the projects controller. This worked for the create and index actions, but I ended up creating a tasks controller because the PUT is handling an individual task and not a project. I thought it was a better way to query the task by itself, instead of querying the project id.

I suppose it would have been more consistent to not use a _member_ route for the GET, but I like that it's concise and keeps other member routes organized under itself.